pkg/lib/cockpit-connect-ssh.tsx

```diff
@@ -0,0 +1,613 @@
+        else if (both)
+            statement = <p>{cockpit.format(_("The SSH key for logging in to $0 is protected. You can log in with either your login password or by providing the password of the key at $1."), this.props.host, this.state.identity_path)}</p>;
+
+        let ssh_key_text = null;
+        let ssh_key_details = null;
+        if (this.state.default_ssh_key) {
+            const key = this.state.default_ssh_key.name;
+            const luser = this.state.user!.name;
+            const lhost = "localhost";
+            const afile = "~/.ssh/authorized_keys";
+            const ruser = this.props.user || address.user || this.state.user!.name;
+            if (!this.state.default_ssh_key.exists) {
+                ssh_key_text = _("Create a new SSH key and authorize it");
+                ssh_key_details = <>
+                    <p>{cockpit.format(_("A new SSH key at $0 will be created for $1 on $2 and it will be added to the $3 file of $4 on $5."), key, luser, lhost, afile, ruser, address.address)}</p>
+                    <FormGroup label={_("Key password")}>
+                        <TextInput id="login-setup-new-key-password" onChange={(_event, value) => this.setState({ login_setup_new_key_password: value })}
+                                type="password" value={this.state.login_setup_new_key_password} />
+                    </FormGroup>
+                    <FormGroup label={_("Confirm key password")}>
+                        <TextInput id="login-setup-new-key-password2" onChange={(_event, value) => this.setState({ login_setup_new_key_password2: value })}
+                                type="password" value={this.state.login_setup_new_key_password2} validated={this.state.login_setup_new_key_password2_error ? "error" : "default"} />
+                        <FormHelper helperTextInvalid={this.state.login_setup_new_key_password2_error} />
+                    </FormGroup>
+                </>;
+            } else {
+                ssh_key_text = _("Authorize SSH key");
+                ssh_key_details = <p>{cockpit.format(_("The SSH key $0 of $1 on $2 will be added to the $3 file of $4 on $5."), key, luser, lhost, afile, ruser, address.address)}</p>;
+            }
+        }
+
+        const body = <>
+            {statement}
+            <br />
+            {(offer_login_password || offer_key_password) &&
+                <Form isHorizontal onSubmit={prevent_default(this.login)}>
+                    {both &&
+                        <FormGroup label={_("Authentication")} isInline hasNoPaddingTop>
+                            <Radio name="auth-method"
+                                   isChecked={this.state.auth === "password"}
+                                   onChange={() => this.setState({ auth: "password" })}
```

Comment:
This added line is not executed by any test.

Reply:
We do test this radio button, but not with switching back and forth. I'll let that slide.